src/otx/config/base/default.yaml

```diff
@@ -0,0 +1,22 @@
+# OTX task enum
```

Comment:
config folder is mainly focus on detection. We will need to find a way to accommodate configs for other tasks.
My suggestion is to structure folder like this.  

```
└── configs
    ├── lightning
    │   ├── anomaly
    │   │   ├── models
    │   │   │   ├── padim.yaml
    │   │   │   ├── fastflow.yaml
    │   │   │   └── ...
    │   │   ├── optimizers
    │   │   │   ├── adam.yaml
    │   │   │   ├── sgd.yaml
    │   │   │   └── ...
    │   │   ├── data
    │   │   │   └── mvtec.yaml
    │   │   ├── deploy
    │   │   │   └── ...
    │   │   └── default.yaml
    │   └── visual_prompt
    │       ├── models
    │       │   └── ...
    │       ├── data
    │       │   └── ...
    │       └── deploy
    │           └── ...
    └── mmengine
        ├── mmseg
        │   ├── models
        │   │   ├── litehrnet.yaml
        │   │   ├── segnext.yaml
        │   │   └── ...
        │   ├── optimizers
        │   │   ├── adam.yaml
        │   │   ├── sgd.yaml
        │   │   └── ...
        │   ├── data
        │   │   ├── coco.yaml
        │   │   ├── datumaro.yaml
        │   │   └── ...
        │   └── deploy
        │       └── ...
        ├── hooks
        │   ├── early_stopping.yaml
        │   ├── ema.yaml
        │   ├── logger.yaml
        │   └── checkpoint.yaml
        └── default.yaml
```

Reply:
It is not specialized for mmdet. Rather than your thought, it has capability with the overriding to support various tasks and frameworks. Let see the following example for multi class classification task:

- `train.yaml`
```yaml
defaults:
...
- model: mmcls
...
```

- `model/multi_class_cls.yaml`
```yaml
...
task: multi_class_cls
...
```

- `model/mmcls.yaml`
```yaml
defaults:
- model/multi_class_cls

model:
 otx_model:
   _target_: otx.model.entity.multi_class_cls.MMCLSCompatibleModel

```

- `recipe/multi_class_cls/some_awesome_model.yaml`
```yaml
defaults:
- model/multi_class_cls

model:
   otx_model:
      config:
         ...
```

Reply:
It appears that we are overriding certain elements in the YAML file. However, the relationship and logic behind these overrides remain unclear, even after reviewing the provided example. Ideally, the overall structure should be straightforward and intuitive.

It might be just my perspective, but even after the example, it's still not clear which YAML files need to be implemented to add a new task. Additionally, the process of customizing my model through this structure is not well-defined.

Reply:
Why do we use Hydra if we don't use YAML file overriding?

```console
└── configs
    ├── lightning
    │   ├── anomaly
    │   │   ├── models
    │   │   │   ├── padim.yaml
    │   │   │   ├── fastflow.yaml
    │   │   │   └── ...
    │   │   ├── optimizers
    │   │   │   ├── adam.yaml
    │   │   │   ├── sgd.yaml
    │   │   │   └── ...
    │   │   ├── data
    │   │   │   └── mvtec.yaml
    │   │   ├── deploy
    │   │   │   └── ...
    │   │   └── default.yaml
    │   └── visual_prompt
    │       ├── models
    │       │   └── ...
    │       ├── data
    │       │   └── ...
    │       └── deploy
    │           └── ...
    └── mmengine
        ├── mmseg
        │   ├── models
        │   │   ├── litehrnet.yaml
        │   │   ├── segnext.yaml
        │   │   └── ...
        │   ├── optimizers
        │   │   ├── adam.yaml
        │   │   ├── sgd.yaml
        │   │   └── ...
        │   ├── data
        │   │   ├── coco.yaml
        │   │   ├── datumaro.yaml
        │   │   └── ...
        │   └── deploy
        │       └── ...
        ├── hooks
        │   ├── early_stopping.yaml
        │   ├── ema.yaml
        │   ├── logger.yaml
        │   └── checkpoint.yaml
        └── default.yaml
```

In your suggestion, many recurring structure happens just like the OTX v1. That why we want to avoid from it and go to OTX v2. It's about the core concepts, so I resolved it. See my below comment.